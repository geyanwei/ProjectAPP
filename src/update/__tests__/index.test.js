let HotUpdateIdx = require.requireActual('../index.js');
let HotUpdate = require.requireMock('react-native').NativeModules.HotUpdate;
jest.mock('../../http/HttpTool.js', () => {
	let json = {
		data: {
			updates: true,
			package: "com.android.apin",
			pname: '1.0.3-rc',
			phash: 'hash',
			descs: '添加聊天功能\n修复商城页面BUG',
			isup: 1,
			url: 'http://update-packages.reactnative.cn/hash',
			pdiffUrl: 'http://update-packages.reactnative.cn/hash', // un-defined
			diffUrl: 'http://update-packages.reactnative.cn/hash' // un-defined
		},
		code: 1
	};
	function post(api, resolve, reject, params) {
		resolve(1, 'success', json, {});
	}
	return { post: post };
});

describe('test hot update index', () => {
	it('Assert isFirstTime to be TRUE', () => {
		expect(HotUpdateIdx.isFirstTime).toMatchSnapshot();
	});
	it('Assert isRolledBack to be FALSE', () => {
		expect(HotUpdateIdx.isRolledBack).toMatchSnapshot();
	});
	it('Assert packageVersion to be 1.0', () => {
		expect(HotUpdateIdx.packageVersion).toMatchSnapshot();
	});
	it('Assert currentVersion to be empty', () => {
		expect(HotUpdateIdx.currentVersion).toEqual('');
	});
	it('Assert getOption exists', () => {
		expect(HotUpdateIdx.getOption()).toMatchSnapshot();
	});
	it('Assert checkUpdate return data', () => {
		expect(HotUpdate.checkUpdate({ phash: '123' })).resolves.toMatchSnapshot();
	});
	it('Assert checkUpdate return error with empty param', () => {
		expect(HotUpdate.checkUpdate()).rejects.toMatchSnapshot();
	});
	it('Assert Success Download return hash', async () => {
		const data = {
				updates: true,
				pname: '1.0.3-rc',
				phash: 'hash',
				descs: '添加聊天功能\n修复商城页面BUG',
				isup: 1,
				url: 'http://update-packages.reactnative.cn/hash',
				pdiffUrl: 'http://update-packages.reactnative.cn/hash', // un-defined
				diffUrl: 'http://update-packages.reactnative.cn/hash' // un-defined
			};
		expect(HotUpdateIdx.downloadUpdate(data)).resolves.toEqual(data.phash);
	});
});
