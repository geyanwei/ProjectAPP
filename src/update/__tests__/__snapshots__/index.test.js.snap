// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test hot update index Assert Success Download return hash 1`] = `
Object {
  "code": 1,
  "data": Object {
    "descs": "添加聊天功能
修复商城页面BUG",
    "diffUrl": "http://update-packages.reactnative.cn/hash",
    "isup": 1,
    "pdiffUrl": "http://update-packages.reactnative.cn/hash",
    "phash": "hash",
    "pname": "1.0.3-rc",
    "updates": true,
    "url": "http://update-packages.reactnative.cn/hash",
  },
}
`;

exports[`test hot update index Assert Success Download return hash 2`] = `undefined`;

exports[`test hot update index Assert getOption exists 1`] = `
Object {
  "package": "com.android.apin",
  "phash": "",
  "type": 1,
  "vcode": "1.0",
}
`;

exports[`test hot update index Assert isFirstTime to be TRUE 1`] = `true`;

exports[`test hot update index Assert isRolledBack to be FALSE 1`] = `false`;

exports[`test hot update index Assert packageVersion to be 1.0 1`] = `"1.0"`;
